File: packages/zako3-data/src/schemas/guild.ts
import { z } from 'zod';

export const guildSummarySchema = z.object({
    guildId: z.string(),
    guildName: z.string(),
    guildIconUrl: z.string().optional(),
    activeChannelId: z.string().optional(),
    activeChannelName: z.string().optional(),
    canManage: z.boolean(),
});

export type GuildSummaryDto = z.infer<typeof guildSummarySchema>;

File: crates/types/src/hq/guild.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, zod_gen_derive::ZodSchema)]
#[serde(rename_all = "camelCase")]
pub struct GuildSummaryDto {
    pub guild_id: String,
    pub guild_name: String,
    pub guild_icon_url: Option<String>,
    pub active_channel_id: Option<String>,
    pub active_channel_name: Option<String>,
    pub can_manage: bool,
}
